### PR TITLE
Update to go 1.11 and fix new linting errors

### PR DESCRIPTION
**Pull request by mboersma**
**What this PR does / why we need it**:

Updates the go compiler used in the go-dev tools image and CircleCI to the [newest release](https://golang.org/doc/go1.11) and fixes the new `go vet` errrors it found.

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
Update to go 1.11 and fix new linting errors
```

### CODE REVIEW

**Comment by mboersma** [comment]
Apparently `gofmt -s` and `goimports` now disagree about the level of indentation for some struct fields, so linting won't pass because they can't both be satisfied. I'll work it out or disable one of the linters.

File: pkg/helpers/helpers_test.go [hunk 1]
```diff
@@ -174,12 +120,12 @@ EPDesL0rH+3s1CKpgkhYdbJ675GFoGoq+X21QaqsdvoXmmuJF9qq9Tq+JaWloUNq
 	pemBuffer := bytes.Buffer{}
 	pem.Encode(&pemBuffer, pemBlock)
 
-	if pemBuffer.String() != expectedPrivateKeyString {
-		t.Fatalf("Private Key did not match expected format/value")
+	if !strings.HasPrefix(pemBuffer.String(), "-----BEGIN RSA PRIVATE KEY-----") {
```
Comment (tariq1890) [inline pkg/helpers/helpers_test.go]: Can we add lines that test the following for the Private Key?

1. privateKey.N.BitLen() equals the SSHKey bit size (which is set at 4096 for us) 
2. privatKey.Validate() does not throw an err

Comment (mboersma) [reply pkg/helpers/helpers_test.go]: Good call--I added that.

**Comment by mboersma** [comment]
This passed all CircleCI jobs previously, and the recent commit just added two unit test clauses, so I think this is good to merge.

**Pull request merged**